src/main/java/org/trigon/hardware/rev/spark/io/SimulationSparkIO.java

```diff
@@ -63,51 +75,63 @@ public void setInverted(boolean inverted) {
     }
 
     @Override
-    public void enableVoltageCompensation(double voltage) {
-        motor.enableVoltageCompensation(voltage);
+    public void setBrake(boolean brake) {
+        SparkMaxConfig configuration = new SparkMaxConfig();
+        configuration.idleMode(brake ? SparkMaxConfig.IdleMode.kBrake : SparkMaxConfig.IdleMode.kCoast);
+        motor.configure(configuration, SparkBase.ResetMode.kNoResetSafeParameters, SparkBase.PersistMode.kNoPersistParameters);
     }
 
     @Override
-    public void setClosedLoopRampRate(double rampRate) {
-        motor.setClosedLoopRampRate(rampRate);
-    }
+    public void updateSimulation() {
+        if (physicsSimulation == null)
+            return;
 
-    @Override
-    public void setSmartCurrentLimit(int limit) {
-        motor.setSmartCurrentLimit(limit);
-    }
+//        physicsSimulation.setInputVoltage(motorSimulation.getAppliedOutput());
+        physicsSimulation.setInputVoltage(motor.getBusVoltage());
```

Comment:
you want the motorsim's bus voltage / applied output, test with it. 
bus voltage will just always give you 12...